hybrid/layout/pv_design_utils.py

```diff
@@ -0,0 +1,316 @@
+    modules_per_string: float,
+    module_power: float,
+    inverter_power: float,
+    n_inputs_inverter: float=50
+    ):
+    """
+    Find the number of strings per inverter to best match target dc_ac_ratio
+    """
+    n_strings_frac = target_system_capacity / (modules_per_string * module_power)
+    n_strings = max(1, round(n_strings_frac))
+
+    if target_dc_ac_ratio < 0:
+        target_dc_ac_ratio = 1
+    n_inverters = find_inverter_count(
+        dc_ac_ratio=target_dc_ac_ratio,
+        modules_per_string=modules_per_string,
+        n_strings=n_strings,
+        module_power=module_power,
+        inverter_power=inverter_power,
+        )
+
+    # Ensure there are enough enough inverters for the number of field connections
+    # TODO: implement get_n_combiner_boxes() and/or string inverter calculations to compute n_field_connections
+    n_field_connections = 1
+    while math.ceil(n_field_connections / n_inverters) > n_inputs_inverter:
+        n_inverters += 1
+
+    # Verify sizing was close to the target size, otherwise error out
+    calculated_system_capacity = verify_capacity_from_electrical_parameters(
+        system_capacity_target=target_system_capacity,
+        n_strings=n_strings,
+        modules_per_string=modules_per_string,
+        module_power=module_power
+    )
+
+    return n_strings, n_inverters, calculated_system_capacity
+
+
+def verify_capacity_from_electrical_parameters(
+    system_capacity_target: float,
+    n_strings: float,
+    modules_per_string: float,
+    module_power: float,
+    ) -> float:
+    """
+    Compute system capacity from specified number of strings, modules per string and module power.
+    If computed capacity is significantly different than the specified capacity an exception will be thrown.
+    
+    inputs:
+        system_capacity_target      [kW]
+        n_strings                   [-]
+        modules_per_string          [-]
+        module_power                [kW]
+
+    returns:
+        calculated_system_capacity  [kW]
+    """
+    PERCENT_MAX_DEVIATION = 5       # [%]
+    calculated_system_capacity = n_strings * modules_per_string * module_power
+    if abs((calculated_system_capacity / system_capacity_target - 1)) * 100 > PERCENT_MAX_DEVIATION:
+        raise Exception(f"The specified system capacity of {system_capacity_target} kW is more than " \
+                        f"{PERCENT_MAX_DEVIATION}% from the value calculated from the specified number " \
+                        f"of strings, modules per string and module power ({int(calculated_system_capacity)} kW).")
+
+    return calculated_system_capacity
+
+
+def align_from_capacity(
+    system_capacity_target: float,
+    modules_per_string: float,
+    module_power: float,
+    inverter_power: float,
+    n_inverters_orig: float,
+    ) -> list:
+    """
+    Ensure coherence between parameters for detailed PV model (pvsamv1),
+    keeping the dc/ac ratio approximately the same
+
+    inputs:
```

Comment:
please check the formatting of the docstring

Reply:
Fixed